### PR TITLE
Change coverage flags to use a per-fn [AtomicUsize;_ ]

### PR DESCRIPTION
**Pull request by llogiq**
This is a thing I wanted to do for some time, and I finally got around to doing it :slightly_smiling_face:. The solution here is to first create a `static <genkey#N>: [AtomicUsize; 0]`, then do all the mutations, come back and patch up the length.

### CODE REVIEW

File: plugin/src/lib.rs [hunk 1]
```diff
@@ -397,10 +437,20 @@ impl<'a, 'cx> MutatorPlugin<'a, 'cx> {
                     ],
                 );
                 quote_expr!(self.cx(), {
-                    ::mutagen::report_coverage($n...$current);
+                    ::mutagen::report_coverage($n..$current, &$sym[$flag], $mask);
                     ::mutagen::ge($right, $left, $n)
                 })
             }
+            BinOpKind::Add => {
```
Comment (gnieto) [inline plugin/src/lib.rs]: Adding this new mutation is intentional?

File: src/coverage.rs [hunk 2]
```diff
@@ -2,41 +2,28 @@ use std::env;
 use std::fs::OpenOptions;
 use std::io::Write;
 use std::ops::Range;
-use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::atomic::{AtomicUsize, Ordering};
 
-lazy_static! {
-    static ref COVERAGE_ALREADY_CHECKED: Vec<AtomicBool> = {
-        let max = env::var("MUTAGEN_MUTATION_AMOUNT").map(|s|s.parse().unwrap_or(0)).unwrap_or(0) + 1;
```
Comment (gnieto) [inline src/coverage.rs]: On the runner, we are still passing this envvar. We should remove this from the runner code.

File: plugin/src/lib.rs (hunk 1 above)
Comment (llogiq) [reply plugin/src/lib.rs]: Yes, I wanted to see if it works with opportunistic mutations. I'm going to extend the build script to generate code for all of them to be included in this place next.

File: src/coverage.rs (hunk 2 above)
Comment (llogiq) [reply src/coverage.rs]: Agreed. :+1:

**Pull request merged**